src/components/DatePickerDocked/DatePickerDockedHeader.tsx

```diff
@@ -0,0 +1,140 @@
+import { memo, useCallback, useMemo } from 'react';
+import type { ViewStyle } from 'react-native';
+
+import { useComponentStyles, useMolecules } from '../../hooks';
+import type { DisableWeekDaysType } from '../DatePickerInline/dateUtils';
+import DayNames, { dayNamesHeight } from '../DatePickerInline/DayNames';
+
+const buttonContainerHeight = 56;
+const buttonContainerMarginTop = 4;
+const buttonContainerMarginBottom = 8;
+
+export type CalendarHeaderProps = {
+    locale?: string;
+    scrollMode: 'horizontal' | 'vertical';
+    onPrev: () => any;
+    onNext: () => any;
+    disableWeekDays?: DisableWeekDaysType;
+    style?: ViewStyle;
+    year: number;
+    onPressYear: (year: number) => void;
+    selectingYear: boolean;
+    selectingMonth: boolean;
+};
+
+function DatePickerDockedHeader({
+    locale = 'en',
+    onPrev,
+    onNext,
+    disableWeekDays,
+    style: styleProp,
+    year,
+    onPressYear,
+    selectingYear,
+    selectingMonth
+}: CalendarHeaderProps) {
+    const { IconButton, View, TouchableRipple, Text } = useMolecules();
+    const componentStyles = useComponentStyles('DatePicker_DockedHeader', styleProp);
+
+    const {
+        containerStyle,
+        buttonContainerStyle,
+        buttonWrapperStyle,
+        spacerStyle,
+        yearButtonStyle,
+        yearInnerStyle,
+        yearLabelStyle,
+        iconContainerStyle,
+        daysWrapperStyle,
+    } = useMemo(() => {
+        const {
+            datePickerHeader,
+            buttonContainer,
+            buttonWrapper,
+            spacer,
+            yearButtonStyle,
+            yearInnerStyle,
+            yearLabelStyle,
+            daysWrapperStyle,
+            ...rest
+        } = componentStyles;
+
+        return {
+            containerStyle: [datePickerHeader, rest],
+            buttonContainerStyle: buttonContainer,
+            buttonWrapperStyle: buttonWrapper,
+            spacerStyle: spacer,
+            yearButtonStyle: [yearButtonStyle, { alignSelf: 'center' }],
+            yearInnerStyle: [yearInnerStyle, { paddingLeft: 0 }],
+            yearLabelStyle,
+            iconContainerStyle: { opacity: 1 },
+            daysWrapperStyle
+        };
+    }, [componentStyles]);
+
+    const onPressDropdown = useCallback(() => onPressYear(year), [onPressYear, year]);
+
+    return (
+        <View style={containerStyle} pointerEvents={'box-none'}>
```

Comment:
I'm not sure why pointerEvents was necessary

Reply:
if I delete pointer events than I am unable to click month picker and year picker buttons
@thetaungg